packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2551,3 +2780,25 @@ class AnnotatedRegionLayer<T extends Object> extends ContainerLayer {
     properties.add(DiagnosticsProperty<bool>('opaque', opaque, defaultValue: false));
   }
 }
+
+/// A utility for managing a layer maintaining a handle to it.
+///
+/// This class manages automatically creation and diposing of layer handles for
+/// its associated [layer].
+class LayerHolder<T extends Layer?> {
```

Comment:
reviewers: I'm somewhat tempted to make this a `mixin` on `RenderObject` so it can override and handle `dispose` for you.  The problem is that it would mean a RenderObject could only easily handle one additional layer, and we'd have to give the layer a weird property name like `additionalLayer` or something.

The framework only needs one additional layer for any of the RenderObjects that tests have picked up, but I'm a bit put off by the naming weirdness and the odd restriction. I'm open to ideas though!

Reply:
What if `LayerHandle` instead of a `dispose` method rendering it unusable simply had a `set layer` that accepted `null`? Then the `LayerHandle` wouldn't need to be nulled out in various `dispose` methods, and we wouldn't need `LayerHolder`?

Strawman:

```dart
class LayerHandle {
  set layer(Layer? layer) {
    if (identical(layer, _layer)) return;
    _layer._unref();
    _layer = layer;
  }
  Layer get layer => _layer!;
}
```

Instead of this:

```dart
      child._parentHandle!.dispose();
      child._parentHandle = null;
```

One would do:

```dart
    child._parentHandle.layer = null;
```

Render objects could use `LayerHandle` and `LayerHolder` wouldn't be necessary.

Reply:
I've more or less done this, but made it `Layer? get layer => _layer;`.